prusti-viper/src/encoder/vir/program.rs

```diff
@@ -0,0 +1,94 @@
+use super::*;
+use prusti_interface::config;
+use viper::{self, AstFactory};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Program {
+    pub fields: Vec<Field>,
+    pub builtin_methods: Vec<BodylessMethod>,
+    pub methods: Vec<CfgMethod>,
+    pub functions: Vec<Function>,
+    pub viper_predicates: Vec<Predicate>,
+}
+
+impl Program {
+    pub fn optimized(mut self) -> Self {
```

Comment:
Could you move
```rust
impl Program {
    pub fn optimized(...) -> Self { ...}
}
```
to the `mod.rs` of the `optimizations` module? Such that `vir` doesn't use stuff from `optimizations`.